traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html

```diff
@@ -171,7 +171,7 @@ <h3>Current Value</h3>
                         </div>
                         </label>
                         <div class="col-md-10 col-sm-10 col-xs-12">
-                            <select id="cdn" name="cdn" class="form-control" ng-model="deliveryService.cdnId" ng-options="cdn.id as cdn.name for cdn in cdns" required>
+                            <select id="cdn" name="cdn" class="form-control" ng-model="deliveryService.cdnId" ng-options="cdn.id as cdn.name for cdn in cdns" required ng-disabled="!settings.isNew">
```

Comment:
i guess you should leave this enabled as you can change the cdn on a DS IF the ds has no ssl keys...

Reply:
The reason I added it because of this statement, "_If the intent is to make CDN immutable on a DS, the cdn field of an existing DS should be disabled as well to prevent the user from changing it._"

Reply:
where did u see that statement?

Reply:
i _think_ cdn and routing name are only immutable if the ds has ssl keys, otherwise, make the change...

Reply:
as a user, i would be annoyed if i created a delivery service and put it in the wrong cdn and was blocked from changing it..

Reply:
Under **_Anything Else_** section in issue: https://github.com/apache/trafficcontrol/issues/5341

Reply:
I'll change it. and make it immutable only if DS has ssl key.

Reply:
yeah, i guess that was a bad assumption on my part. I _think_ if a ds has no ssl keys, then changing the cdn and/or routing name is perfectly fine.